pkg/operator/controller/controller_dns_configmap.go

```diff
@@ -344,6 +356,28 @@ func coreDNSLogLevel(dns *operatorv1.DNS) string {
 	return "class error"
 }
 
+// coreDNSCache reads the TTL values set in spec.cache and returns integer representations of those values in seconds.
+// This returns uint32 values because CoreDNS won't allow setting the maxTTLs to 0 and there's no use case for setting a
+// negative maxTTL value.
+func coreDNSCache(dns *operatorv1.DNS) (positiveTTL, negativeTTL uint32) {
```

Comment:
Would it be better to make this func more "functional"? So a function which takes in values and returns value and doesn't need to operate on an instance of `*operatorv1.DNS`. It's arguably easier to test free-form functions.

Reply:
If its a blocker I can change it. The intent with the function argument and name was to follow the same pattern as `coreDNSLogLevel()`, `coreDNSPolicy()`, etc. This one is nuanced comparitively since it returns two values but the scope is the same; taking the `DNSSpec`, finding the right configuration, then returning a Corefile acceptable value (or set of values).

Reply:
I don't see the problem with taking a `*operatorv1.DNS`.  It's still a pure function, and having the function take `*operatorv1.DNS` as input means that the unit test for the function uses test input that is in the same format as what we get from the API.  

Using named return values is unusual in this codebase, but I won't hold that against you.  * grin *.